src/auth.ts

```diff
@@ -0,0 +1,208 @@
+import Messenger from './messenger';
+import { AuthOptions, EmitterMessage, WorkerMessage } from './types';
+
+export default class Auth {
+  // Options
+  private clientId: string;
+  private disableRedirectClose: boolean;
+  private isDevelopment: boolean;
+  private onTokenFetched?: (address: string) => void;
+  private redirectUrl?: string;
+
+  // Config
+  private host: string;
+  private messenger: Messenger;
+
+  // State
+  public isLoggingIn: boolean;
+  public selectedAddress: string;
+  private didHandleRedirect = false;
+
+  constructor({
+    authToken,
+    clientId,
+    disableRedirectClose,
+    host,
+    messenger,
+    redirectURL,
+    _isDevelopment,
+    _onTokenFetched,
+  }: AuthOptions) {
+    this.clientId = clientId;
+    this.disableRedirectClose = !!disableRedirectClose;
```

Comment:
`disableRedirectClose` is an optional param - so it can be `undefined` - the class member is a `boolean`, so it's gotta get converted - it's a little safer/cleaner imo to pass around a boolean. Can change if you have strong opinions 🤷

Reply:
No opinions here, I just wanted to make sure it was the same thing I was thinking it was for.